plugins/kotlin/frontend-independent/src/org/jetbrains/kotlin/idea/findUsages/KotlinFindUsagesSupport.kt

```diff
@@ -29,14 +34,18 @@ interface KotlinFindUsagesSupport {
         fun PsiReference.isConstructorUsage(ktClassOrObject: KtClassOrObject): Boolean =
             getInstance(ktClassOrObject.project).isConstructorUsage(this, ktClassOrObject)
 
-        fun getSuperMethods(declaration: KtDeclaration, ignore: Collection<PsiElement>?) : List<PsiElement> =
+        fun getSuperMethods(declaration: KtDeclaration, ignore: Collection<PsiElement>?): List<PsiElement> =
             getInstance(declaration.project).getSuperMethods(declaration, ignore)
 
         fun sourcesAndLibraries(delegate: GlobalSearchScope, project: Project): GlobalSearchScope =
             getInstance(project).sourcesAndLibraries(delegate, project)
     }
 
-    fun isCallReceiverRefersToCompanionObject(element: KtElement, companionObject: KtObjectDeclaration): Boolean
+    fun <T> analyseIfCallReceiverRefersToCompanionObject(
```

Comment:
Name and semantic of the function is not very clear. Let's just move and make abstract to the `KotlinFindUsagesSupport` the whole `KotlinFindClassUsagesHandler.processCompanionObjectInternalReferences`